samcli/cli/main.py

```diff
@@ -49,11 +50,115 @@ def print_info(ctx, param, value):
     if not value or ctx.resilient_parsing:
         return
 
-    click.echo(json.dumps({"version": __version__}, indent=2))
+    info = {
+        "version": __version__,
+        "system": _gather_system_info(),
+        "additional_dependencies": _gather_additional_dependencies_info(),
+    }
+    click.echo(json.dumps(info, indent=2))
 
     ctx.exit()
 
 
+def _gather_system_info() -> Dict[str, str]:
+    """
+    Gather system info
+
+    Returns
+    -------
+    dict[str, str]
+    """
+    from platform import platform, python_version
+
+    info = {
+        "python": python_version(),
```

Comment:
did you try to run `sam --info` for the different ways of installations, specially if there is no python installed on the machine?

Reply:
`sam` command is run on the Python interpreter installed along with SAM CLI. So it still works even if a user hasn't installed Python on their own.

Reply:
Also confirmed we have existing integration test for `sam --info` already - https://github.com/aws/aws-sam-cli/blob/develop/tests/integration/root/test_root_command.py#L13